File: boardgametracker.client/src/models/ScoreSheet/ScoreSession.ts
export interface ScoreSession {
  id?: number;
  name: string;
  scoreSheetTemplateId: number;
  templateVersionSnapshot: string;
  jsonData: string;
  gameId?: number;
  locationId?: number;
  createdByUserId: string;
  createdAt: Date;
  updatedAt: Date;
  finishedAt?: Date;
  notes: string;
  isCompleted: boolean;
}

export interface ScoreSessionData {
  players: ScorePlayer[];
  rounds?: ScoreRound[];
  fieldValues: Record<string, Record<string, any>>; // playerId -> fieldId -> value
  totals: Record<string, number>; // playerId -> total
}

export interface ScorePlayer {
  id: string;
  name: string;
  order: number;
}

export interface ScoreRound {
  id: string;
  name: string;
  order: number;
  fieldValues: Record<string, Record<string, any>>; // playerId -> fieldId -> value
}

File: BoardGameTracker.Common/Entities/ScoreSession.cs
using BoardGameTracker.Common.Entities.Helpers;

namespace BoardGameTracker.Common.Entities;

public class ScoreSession : HasId
{
    public string Name { get; set; }
    public int ScoreSheetTemplateId { get; set; }
    public ScoreSheetTemplate ScoreSheetTemplate { get; set; }
    public string TemplateVersionSnapshot { get; set; }
    public string JsonData { get; set; }
    public int? GameId { get; set; }
    public Game Game { get; set; }
    public int? LocationId { get; set; }
    public Location Location { get; set; }
    public string CreatedByUserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? FinishedAt { get; set; }
    public string Notes { get; set; }
    public bool IsCompleted { get; set; } = false;
}